android/modules/ui/src/java/ti/modules/titanium/ui/ShortcutModule.java

```diff
@@ -6,21 +6,239 @@
 @Kroll.module(parentModule = UIModule.class)
 public class ShortcutModule extends KrollModule
 {
 	private static final String TAG = "ShortcutModule";
 
+	// Only supported on Android 7.1 and above.
+	private static final boolean shortcutCapability = Build.VERSION.SDK_INT >= Build.VERSION_CODES.N_MR1;
+
+	private static ShortcutManager shortcutManager = null;
+
 	public ShortcutModule()
 	{
 		super("Shortcut");
+
+		if (!shortcutCapability) {
+			return;
+		}
+
+		// Obtain `ShortcutManager` instance.
+		if (shortcutManager == null) {
+			final Context context = TiApplication.getInstance();
+			shortcutManager = (ShortcutManager) context.getSystemService(Context.SHORTCUT_SERVICE);
+		}
+	}
+
+	/**
+	 * Create a `ShortcutItemProxy` array from a `ShortcutInfo` list.
+	 * @param shortcuts List of `ShortcutInfo` objects.
+	 * @return Array of `ShortcutItemProxy` objects.
+	 */
+	@RequiresApi(api = Build.VERSION_CODES.N_MR1)
+	private ShortcutItemProxy[] toShortcutItems(List<ShortcutInfo> shortcuts)
+	{
+		final List<ShortcutItemProxy> items = new ArrayList<>();
+
+		for (ShortcutInfo shortcut : shortcuts) {
+			final ShortcutItemProxy shortcutItemProxy = new ShortcutItemProxy();
+			final KrollDict creationDict = new KrollDict();
+
+			// Translate into ShortcutItemProxy.
+			creationDict.put(TiC.PROPERTY_ID, shortcut.getId());
+			creationDict.put(TiC.PROPERTY_TITLE, shortcut.getShortLabel());
+			creationDict.put(TiC.PROPERTY_DESCRIPTION, shortcut.getLongLabel());
+
+			// Obtain additional stored data.
+			if (shortcut.getExtras() != null) {
+				final PersistableBundle bundle = shortcut.getExtras();
+
+				if (bundle.containsKey(TiC.PROPERTY_ICON)) {
+					creationDict.put(TiC.PROPERTY_ICON, bundle.get(TiC.PROPERTY_ICON));
+				}
+				if (bundle.containsKey(TiC.PROPERTY_DATA)) {
+					try {
+						final KrollDict data = new KrollDict(new JSONObject(bundle.getString(TiC.PROPERTY_DATA)));
+						creationDict.put(TiC.PROPERTY_DATA, data);
+					} catch (JSONException e) {
+						Log.w(TAG, "Could not parse bundle data property.");
+					}
+				}
+			}
+
+			// Initialize ShortcutItemProxy.
+			shortcutItemProxy.handleCreationDict(creationDict);
+			items.add(shortcutItemProxy);
+		}
+
+		return items.toArray(new ShortcutItemProxy[items.size()]);
+	}
+
+	/**
+	 * Obtain array of current active shortcuts.
+	 * @return Array of `ShortcutItemProxy` objects.
+	 */
+	@Kroll.getProperty
+	public ShortcutItemProxy[] items()
+	{
+		if (!shortcutCapability || shortcutManager == null) {
+			return null;
+		}
+
+		final List<ShortcutInfo> shortcuts = shortcutManager.getDynamicShortcuts();
+		return toShortcutItems(shortcuts);
+	}
+
+	/**
+	 * Obtain array of current active static shortcuts.
+	 * @return Array of `ShortcutItemProxy` objects.
+	 */
+	@Kroll.getProperty
+	public ShortcutItemProxy[] staticItems()
+	{
+		if (!shortcutCapability || shortcutManager == null) {
+			return null;
+		}
+
+		return toShortcutItems(shortcutManager.getPinnedShortcuts());
```

Comment:
I think we need to call `ShortcutManager.getManifestShortcuts()` instead.
https://developer.android.com/reference/android/content/pm/ShortcutManager#getManifestShortcuts()

Google's docs suggest that pinned shortcut list can contain both static and dynamic shortcuts.
https://developer.android.com/guide/topics/ui/shortcuts/creating-shortcuts#pinned